shared/modules/doc/README.md

```diff
@@ -85,6 +106,8 @@ if err != nil {
 
 ##### Add a logger to the module
 
+<!-- DISCUSS: I believe we should change this convention, to me it's more semantic if logging is configured during construction/initialization and not during `Start` (which some modules might not even have if they don't implement `InterruptableModule`). I believe that a better approach is summarized here: https://github.com/pokt-network/pocket/blob/8bee148f3b0e768154be4bce02e94813c9382aac/state_machine/module.go#L29-L32 -->
```

Comment:
Thanks for calling this out @deblasis! :raised_hands: 

It may be slightly tangential but I noticed another seemingly conventional thing that we may want to consider changing:

It looks to me like our `#Create()` **methods** (i.e. not the static pkg-level constructor) return a **new** pointer, essentially ignoring "self". As long as this method's usage is limited to scenarios where the reference to the original receiver of `#Create()` is not shared.

I think my observation can be reduced to, do we expect the following to be true (i.e. pass), because right now it's not:

```golang
package state_machine

import (
	"testing"

	"github.com/golang/mock/gomock"
	"github.com/stretchr/testify/require"

	mockModules "github.com/pokt-network/pocket/shared/modules/mocks"
)

func TestStateMachineModule_Create(t *testing.T) {
	ctrl := gomock.NewController(t)

	busMock := mockModules.NewMockBus(ctrl)
	busMock.EXPECT().PublishEventToBus(gomock.Any()).AnyTimes()
	busMock.EXPECT().RegisterModule(gomock.Any()).AnyTimes()

	mod1 := new(stateMachineModule)

	mod2, err := mod1.Create(busMock)
	require.NoError(t, err)
	require.Equal(t, mod1, mod2)
}
```

To make this true we would need to change our convention in `#Create()` like so:

```
diff --git i/state_machine/module.go w/state_machine/module.go
index b8655084..38bb22e3 100644
--- i/state_machine/module.go
+++ w/state_machine/module.go
@@ -25,8 +25,8 @@ func Create(bus modules.Bus, options ...modules.ModuleOption) (modules.Module, e
        return new(stateMachineModule).Create(bus, options...)
 }
 
-func (*stateMachineModule) Create(bus modules.Bus, options ...modules.ModuleOption) (modules.Module, error) {
-       m := &stateMachineModule{
+func (m *stateMachineModule) Create(bus modules.Bus, options ...modules.ModuleOption) (modules.Module, error) {
+       *m = stateMachineModule{
                logger: logger.Global.CreateLoggerForModule(modules.StateMachineModuleName),
        }
 
```

Reply:
You raise a fair point @bryanchriswhite, I think this should be a separate issue since it doesn't change the signatures of the interfaces.

The idea was that modules should only be instantiated/constructed/initialized from outside their package using the `<package_name>.Create(...)`

This is because we wanted a consistent signature for modules enforced by interfaces but also we wanted to keep the structs private/unexported.

Consequently, this is not a problem because it's not possible to call `new(stateMachineModule).Create(...)` from outside the package since `stateMachineModule` is inaccessible.

To answer your question more directly: I personally don't expect that test to succeed. As in, that's not what I would test for. 
I would test for the fact that the `Create(bus)` sets a bus internally for example.

The `Create` is like a factory on the type and by definition, it "spits out" new instances if that makes sense.

If this sounds confusing even just an inch, let's change this.

Reply:
@Olshansk it's an option but then it means that you can't enforce a `create(...)` signature via the interface.

two_red_buttons_cannot_choose_sweating.gif

Reply:
As a stoic, I pick neither ;)

Reply:
💀

Reply:
@bryanchriswhite I captured the discussion in #642